workflow-service/src/main/java/com/redhat/parodos/workflow/execution/aspect/AssessmentInfrastructureWorkFlowPostInterceptor.java

```diff
@@ -66,16 +72,26 @@ public WorkReport handlePostWorkFlowExecution() {
 				.getWorkFlowTaskDefinitions().stream().map(WorkFlowTaskDefinition::getWorkFlowCheckerMappingDefinition)
 				.filter(Objects::nonNull).collect(Collectors.toSet());
 
-		if (workFlowCheckerMappingDefinitions.stream()
+		List<WorkFlowExecution> checkerExecutions = workFlowCheckerMappingDefinitions.stream()
 				.map(workFlowCheckerDefinition -> workFlowRepository
 						.findFirstByWorkFlowDefinitionIdAndMasterWorkFlowExecution(
 								workFlowCheckerDefinition.getCheckWorkFlow().getId(), masterWorkFlowExecution))
-				.anyMatch(checkerExecution -> checkerExecution == null
-						|| !WorkFlowStatus.COMPLETED.equals(checkerExecution.getStatus()))) {
-			log.info("fail workflow: {} because it has pending/running checker(s)", workFlowDefinition.getName());
-			workFlowExecution.setStatus(WorkFlowStatus.PENDING);
-			report = new DefaultWorkReport(WorkStatus.IN_PROGRESS, workContext);
-		}
+				.collect(Collectors.toList());
+
+		for (WorkFlowExecution checkerExecution : checkerExecutions)
+			if (checkerExecution != null && WorkFlowStatus.REJECTED.equals(checkerExecution.getStatus())) {
```

Comment:
should we add a method in WorkFlowStatus.IsRejected() or WorkFlowStatus.IsFailed()